SRC/CLASSES.CPP

```diff
@@ -282,32 +283,31 @@ void Player::chk_keys( int index )
 
 void Body_part::move()
 {
-    int offs[2*4]; //**
-    int reuna = 4; //**chk pointit on näin
-    int reuna2 = 4; //**
-    int rnx, rny; //**
-    float newx, newy;
-    newx = x + ( ( int ) speed*sini[angle] );
-    newy = y + ( ( int ) speed*cosi[angle] );
-    rnx = newx;
-    rny = newy;
-    if ( num < 12 )
-        if ( rand() % 3 != 0 ) new_effect( 0, BLOOD2, int( x ) + 10, int( y ) + 10, 0, 0 );
-    offs[0] = ( ( rny + reuna )  / 20 ) *level_x_size + ( ( rnx + 10 - reuna2 )  / 20 );
-    offs[1] = ( ( rny + reuna )  / 20 ) *level_x_size + ( ( rnx + 10 + reuna2 )  / 20 );
-    offs[2] = ( ( rny + 20 - reuna )  / 20 ) *level_x_size + ( ( rnx + 10 - reuna2 )  / 20 );
-    offs[3] = ( ( rny + 20 - reuna )  / 20 ) *level_x_size + ( ( rnx + 10 + reuna2 )  / 20 );
-    offs[4] = ( ( rny + 10 - reuna2 )  / 20 ) *level_x_size + ( ( rnx + reuna )  / 20 );
-    offs[5] = ( ( rny + 10 + reuna2 )  / 20 ) *level_x_size + ( ( rnx + reuna )  / 20 );
-    offs[6] = ( ( rny + 10 - reuna2 )  / 20 ) *level_x_size + ( ( rnx + 20 - reuna )  / 20 );
-    offs[7] = ( ( rny + 10 + reuna2 )  / 20 ) *level_x_size + ( ( rnx + 20 - reuna )  / 20 );
-    if ( newy < y && level[offs[0]].type == FLOOR && level[offs[1]].type == FLOOR )
+    int reuna = 4; //  * * chk pointit on näin
+    int reuna2 = 4; // * *
+    if ( num < 12 && rand() % 3 != 0 )
+    {
+        new_effect( 0, BLOOD2, int( x ) + 10, int( y ) + 10, 0, 0 );
+    }
+    float newx = x + ( ( int ) speed*sini[angle] );
+    float newy = y + ( ( int ) speed*cosi[angle] );
+    int rnx = newx;
+    int rny = newy;
+    int type0 = get_block_type_at_coord( rnx + 10 - reuna2, rny + reuna );
```

Comment:
`int types[]`?

Reply:
Useless use of array in my opinion. We never iterate over these, and they're definitely not homogenous values, but each index of the array has a special meaning.